### PR TITLE
trivial: Add a tool to generate the release notes

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/generate-release.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=invalid-name,missing-docstring
+
+import subprocess
+import datetime
+
+
+def _get_last_release() -> str:
+    p = subprocess.run(
```
Comment (superm1) [inline contrib/generate-release.py:14]: subprocess.check_output

File: contrib/generate-release.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+def _generate_release_notes(last_tag: str, next_tag: str) -> str:
+    args = [
+        "git",
+        "log",
+        "--format=%s",
+        "--cherry-pick",
+        "--right-only",
+        f"{last_tag}...",
+    ]
+    p = subprocess.run(args, check=True, capture_output=True)
```
Comment (superm1) [inline contrib/generate-release.py:38]: subprocess.check_output

File: contrib/generate-release.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+def _get_appstream_date() -> str:
+    return datetime.datetime.now().strftime("%Y-%m-%d")
+
+
+def _generate_release_notes(last_tag: str, next_tag: str) -> str:
+    args = [
+        "git",
+        "log",
+        "--format=%s",
+        "--cherry-pick",
```
Comment (superm1) [inline contrib/generate-release.py:34]: Why need this one?

File: contrib/generate-release.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        if line.find("Merge") != -1:
+            continue
+        if line in lines_feat or line in lines_bugs or line in lines_devs:
+            continue
+        if line.startswith("Add "):
+            lines_feat.append(line)
+            continue
+        lines_bugs.append(line)
+
+    out: str = ""
```
Comment (superm1) [inline contrib/generate-release.py:59]: This feels like you reinvented Jinja2

File: contrib/generate-release.py (hunk 1 above)
Comment (hughsie) [reply contrib/generate-release.py:14]: TIL, thanks.

File: contrib/generate-release.py (hunk 2 above)
Comment (hughsie) [reply contrib/generate-release.py:38]: Fixed.

File: contrib/generate-release.py (hunk 3 above)
Comment (hughsie) [reply contrib/generate-release.py:34]: For this case I don't -- not `--right-only` -- both removed.

**Review by superm1 — APPROVED** [review]

**Pull request merged**